src/rules/mixin-no-risky-parent-selectors/index.js

```diff
@@ -0,0 +1,67 @@
+"use strict";
+
+const { utils } = require("stylelint");
+const namespace = require("../../utils/namespace");
+const ruleUrl = require("../../utils/ruleUrl");
+
+const ruleName = namespace("mixin-no-risky-parent-selectors");
+
+const messages = utils.ruleMessages(ruleName, {
+  rejected: `Unexpected nested parent selector in @mixin rule.`
+});
+
+const meta = {
+  url: ruleUrl(ruleName)
+};
+
+function isWithinMixin(node) {
+  let parent = node.parent;
+  while (parent) {
+    if (parent.type === "atrule" && parent.name === "mixin") {
+      return true;
+    }
+    parent = parent.parent;
+  }
+  return false;
+}
+
+function hasNestedParentSelector(selectors) {
+  return selectors
+    .split(",")
+    .some(
+      selector =>
+        selector.includes("&") && !selector.replace(" ", "").startsWith("&")
```

Comment:
I don't think this is quite right. Specifically, this won't produce an error for `& .foo &`, but it should (another test case to add).

Reply:
Ohh right.. thanks for pointing this out!